imager/profiles/models.py

```diff
@@ -18,6 +21,49 @@ class ImagerProfile(models.Model):
     phone_priv = models.BooleanField(default=True)
     birthday_priv = models.BooleanField(default=True)
 
+    follows = models.ManyToManyField("self", symmetrical=False)
+
+    blocked = models.ManyToManyField("self")
+
+    def block(self, IProfile):
+        if IProfile in self.blocked.all():
+            raise ValueError('You been BLOCKED!')
+        self.blocked.add(IProfile)
+
+    def unblock(self, IProfile):
+        if IProfile not in self.blocked.all():
+            raise ValueError('Not in blocked list')
+        for block in self.blocked.all():
+            if block == IProfile:
+                self.blocked.remove(block)
+                break
+
+    def following(self):
+        follow_list = self.follows.all()
+        for follow in follow_list:
+            if follow in self.blocked.all():
+                follow.delete()
```

Comment:
I don't think you want to do this.  It seems like this would result in the deletion of the Profile at the other end of this relationship, not in the deletion of the following relationship.